contract_compassion/model/contract_group.py

```diff
@@ -0,0 +1,258 @@
+        """
+        contract_ids = list()
+        inv_vals = dict()
+        if 'payment_term_id' in vals:
+            inv_vals['payment_term'] = vals['payment_term_id']
+            payment_term = self.pool.get('account.payment.term').browse(
+                cr, uid, vals['payment_term_id'], {'lang': 'en_US'})
+            payment_name = payment_term.name
+            wf_service = netsvc.LocalService('workflow')
+            for group in self.browse(cr, uid, ids, context):
+                old_term = group.payment_term_id.name
+                for contract in group.contract_ids:
+                    contract_ids.append(contract.id)
+                    if 'LSV' in payment_name or 'Postfinance' in payment_name:
+                        wf_service.trg_validate(
+                            uid, 'recurring.contract', contract.id,
+                            'will_pay_by_lsv_dd', cr)
+                        # LSV/DD Contracts need no reference
+                        if group.bvr_reference and \
+                                'multi-months' not in payment_name:
+                            vals['bvr_reference'] = False
+                    elif 'LSV' in old_term or 'Postfinance' in old_term:
+                        wf_service.trg_validate(
+                            uid, 'recurring.contract', contract.id,
+                            'mandate_validated', cr)
+        if 'bvr_reference' in vals:
+            inv_vals['bvr_reference'] = vals['bvr_reference']
+            for group in self.browse(cr, uid, ids, context):
+                contract_ids.extend([c.id for c in group.contract_ids])
+
+        res = super(contract_group, self).write(cr, uid, ids, vals, context)
+
+        if contract_ids:
+            # Update related open invoices to reflect the changes
+            inv_line_obj = self.pool.get('account.invoice.line')
+            inv_obj = self.pool.get('account.invoice')
+            inv_line_ids = inv_line_obj.search(cr, uid, [
+                ('contract_id', 'in', contract_ids),
+                ('state', 'not in', ('paid', 'cancel'))], context=context)
+            invoice_ids = list(set([inv_l.invoice_id.id for inv_l in
+                                    inv_line_obj.browse(cr, uid, inv_line_ids,
+                                                        context)]))
+            inv_obj.action_cancel(cr, uid, invoice_ids, context)
+            inv_obj.action_cancel_draft(cr, uid, invoice_ids)
+            inv_obj.write(cr, uid, invoice_ids, inv_vals, context)
+            wf_service = netsvc.LocalService('workflow')
+            for invoice_id in invoice_ids:
+                wf_service.trg_validate(uid, 'account.invoice', invoice_id,
+                                        'invoice_open', cr)
+        return res
+
+    def clean_invoices(self, cr, uid, group, advance_billing=0, context=None):
+        ''' Override clean_invoices to delete cancelled invoices '''

```

Comment:
Double quotes for docstrings.
https://www.python.org/dev/peps/pep-0257/#one-line-docstrings